folium/folium.py

```diff
@@ -616,10 +618,21 @@ def _popup_render(self, popup=None, mk_name=None, count=None,
         if not popup_on:
             return 'var no_pop = null;'
         else:
-            if isinstance(popup, str):
+            if sys.version_info >= (3,0):
+                utype, stype = str, bytes
+            else:
+                utype, stype = unicode, str
+
+            if isinstance(popup, (utype, stype)):
                 popup_temp = self.env.get_template('simple_popup.js')
+                if isinstance(popup, utype):
+                    popup_txt = popup.encode('ascii', 'xmlcharrefreplace')
+                else:
+                    popup_txt = popup
+                if sys.version_info >= (3,0):
+                    popup_txt = popup_txt.decode()
                 return popup_temp.render({'pop_name': mk_name + str(count),
-                                          'pop_txt': json.dumps(popup),
+                                          'pop_txt': json.dumps(str(popup_txt)),
```

Comment:
@apatil I am on the fence here.  I am not sure if the call to `str` is necessary, but it does not hurt...  What is your argument?